stk/utilities/utilities.py

```diff
@@ -1143,10 +1146,371 @@ def vector_theta(vector1, vector2):
+            'DMSO'.lower(), 'ether', 'H2O'.lower(),
+            'methanol', 'THF'.lower(), 'toluene'
+        ])
+        return solvent in valid_solvents
+    elif gfn_version == 2:
+        valid_solvents = set([
+            'acetone', 'acetonitrile', 'CH2Cl2'.lower(),
+            'CHCl3'.lower(), 'CS2'.lower(), 'DMF'.lower(),
+            'DMSO'.lower(), 'ether', 'H2O'.lower(), 'methanol',
+            'n-hexane'.lower(), 'THF'.lower(), 'toluene'
+        ])
+        return solvent in valid_solvents
+
+
+class XTBExtractor:
+    """
+    Extracts properties from GFN-xTB output files.
+
+    Attributes
+    ----------
+    output_file : :class:`str`
+        Output file to extract properties from.
+
+    """
+    def __init__(self, output_file):
+        """
+        Initializes :class:`XTBExtractor`
+
+        Parameters
+        ----------
+        output_file : :class:`str`
+            Output file to extract properties from.
+
+        """
+        self.output_file = output_file
+        with open(self.output_file, 'r') as f:
+            self.output_string = f.readlines()
+
+    def total_energy(self):
+        """
+        Extracts total energy in a.u. from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`float`
+            Total energy in a.u.
+
+        """
+
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '          | TOTAL ENERGY  ' in line:
+                return float(nums.search(line.rstrip()).group(0))
+
+    def homo_lumo_gap(self):
+        """
+        Extracts HOMO-LUMO gap in eV from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`float`
+            Homo-Lumo gap in eV.
+
+        """
+
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '          | HOMO-LUMO GAP   ' in line:
+                return float(nums.search(line.rstrip()).group(0))
+
+    def fermi_level(self):
+        """
+        Extracts Fermi-level energy in eV from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`float`
+            Fermi-level in eV.
+
+        """
+
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '             Fermi-level        ' in line:
+                part2 = line.split('Eh')
+                return float(nums.search(part2[1].rstrip()).group(0))
+
+    def qonly_dipole_moment(self):
+        """
+        Extracts q only dipole moment in Debye from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of dipole moment as [x, y, z] in a
```

Comment:
two backticks on each side of list

Reply:
probably dont need to state that its length 3 if you are giving the list directly.